src/network/networkd-link.c

```diff
@@ -1490,6 +1490,30 @@ bool link_has_carrier(Link *link) {
         return false;
 }
 
+bool link_should_activate(Link *link) {
+        assert(link);
+
+        if (!link->network)
+                return true;
```

Comment:
hmm, a link for which we have no network file (i.e. that is not managed by us) should not be activated by us. It should be left as it is.

I think this function should probably return a tri-state, i.e. an int that can be < 0, 0 or > 0. And if > 0 is returned this means: "keep me up", and == 0 means "keep me down", and < 0 means "keep me how I am"